src/query/storage/m3/storage.go

```diff
@@ -281,31 +292,32 @@ func (s *m3storage) Close() error {
 	return nil
 }
 
-func (w *writeRequest) Process(ctx context.Context) error {
-	common := w.writeRequestCommon
-	store := common.store
-	id := common.id
-
+func (s *m3storage) writeSingle(
+	ctx context.Context,
+	query *storage.WriteQuery,
+	datapoint ts.Datapoint,
+	identID ident.ID,
+	iterator ident.TagIterator) error {
 	var (
 		namespace ClusterNamespace
 		err       error
 	)
-	switch common.attributes.MetricsType {
+	switch query.Attributes.MetricsType {
```

Comment:
nit: pull out `query.Attributes` and `s.clusters` into variables

Reply:
i think s.clusters is fine